src/main.rs

```diff
@@ -130,6 +132,34 @@ async fn start_servers(
     });
 }
 
+async fn schedule_background_jobs(
+    app_config: &snapchain::cfg::Config,
+    block_store: BlockStore,
+    local_state_store: LocalStateStore,
+) {
+    let sched = JobScheduler::new().await.unwrap();
+    let is_sync_complete = local_state_store.is_sync_complete().unwrap();
+    if app_config.read_node && is_sync_complete {
+        if let Some(block_retention) = app_config.read_node_block_retention {
+            let throttle = tokio::time::Duration::from_millis(100); // TODO: make const or configurable
+            let cutoff_timestamp =
+                util::get_farcaster_time().unwrap() - (block_retention.as_secs() as u64);
```

Comment:
Let's pass in just the block_retention, and have the job calculate the timestamp